### ISSUE TITLE
Tracegen/Hookgen don

### ISSUE THREAD

**Issue by dnakamura** [eclipse/omr#2725]

**Comment by dnakamura** [comment]
Because the hookgen/tracegen tools use the a2e library to open files, it doesn't properly deal with ascii tagged files on z/os. Normally, if autoconversion is enabled, the OS would silently convert the file on the fly. However in the a2e library, autocovnersion is explicitly disabled when it opens files (I assume for the benefit of file I/O from java).

This deficiency in the trace/hook tools means that the trace/hook definitions need to be explictly stored in ebcdic, when building on z/os (see #2721). All this just so that a2e can convert it back to ascii before passing the data on to the program. 

As I see it, there are 2 possible solutions to this issue. 

1.  Compile the tools with the `-qascii` flag. This marks the executables as expecting/producing acsii. This means we can bypass the a2e library all together, and have the heavy lifting done by the os/c-library. (I have this prototyped with #2718). The downside to this approach is that if auto file tagging is enabled (which is required for git),  output files are merely tagged as being in ascii, rather than being converted to ebcdic.  This is an issue for the output of tracemerge, which is expected to be in platform-native encoding.  THis can be made to work properly but clutters up the code base with `#ifdef`s and zos specific code

2.  Have the tooling deal natively with ebcdic. Rather than using ascii for all internal strings just use the native platform encoding (and cut `a2e` out of the loop entirely).  This means that the output will be in ebcdic (as it was before), and ascii input will be handled automatically (assuming the files are properly tagged, and autoconversion is enabled). The downside to this approach is that it does require patching the omr copy of the `pugixml` library to properly handle ebcdic. In addition we will need to build 2 copies of pugixml (which is already done in the makefiles, but not in cmake).

**Comment by dnakamura** [comment]
Anyone have any opinions? I'm leaning toward solution 2, but I am open to feedback

**Comment by rwy0717** [comment]
Why do we need two copies of pugixml? One with a2e (for java), one without (for build tools)? I would also lean towards just using the native encoding.

**Comment by dnakamura** [comment]
yes. one for ascii, one for ebcdic. (although the ascii version may only be used by the fvtests, in which case its possible to convert them as well).

**Comment by dnakamura** [comment]
I should also point out that the patches to `pugixml` would probably not be 100% correct since it makes some assumptions based on UTF-8, however it will work properly as long as you dont use any non-ascii characters.

**Comment by dnakamura** [comment]
Implementation of method 2 can be seen with #2727
